src/veritysetup/veritysetup.c

```diff
@@ -109,7 +140,49 @@ static int parse_options(const char *options) {
                 else if (streq(word, "panic-on-corruption"))
                         arg_activate_flags |= CRYPT_ACTIVATE_PANIC_ON_CORRUPTION;
 #endif
-                else if ((val = startswith(word, "hash-offset="))) {
+                else if ((val = startswith(word, "superblock="))) {
+
+                        r = parse_boolean(val);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse %s: %m", word);
+
+                        arg_superblock = r;
+                } else if ((val = startswith(word, "format="))) {
+                        int i;
+
+                        r = safe_atoi(val, &i);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse %s: %m", word);
+                        if (arg_format > 1)
+                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "format expects either 0 (original Chrome OS version) or "
+                                                                                "1 (current version).");
```

Comment:
current is a bit misleading. That suggests the meaning would change whenever a new verson is invented. But htat's not the case. maybe say the default format as of kernel x.y.z?

Reply:
I am not sure I get it...

However, both formats were introduced since [3.4](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=a4ffc152198efba2ed9e6eac0eb97f17bfebce85).

Here, I am just checking for the format values (as in [cryptsetup](https://gitlab.com/cryptsetup/cryptsetup/-/blob/main/lib/setup.c#L2033-2036)) if one wants to use the format Chrome OS format (aka `0`).

Reply:
Oh, I get it now, let me check for the current version in kernel and cryptsetup maybe...

Reply:
(I strongly guess it has always been format 1 since verity is upstream... I am doing git log to confirm and get back afterward).

Reply:
The kernel documentation says version `1` is default since initial [commit](https://github.com/torvalds/linux/blob/a4ffc152198efba2ed9e6eac0eb97f17bfebce85/Documentation/device-mapper/verity.txt#L15-L24); the verity module takes the version from the ["constructor"](https://github.com/torvalds/linux/blob/a4ffc152198efba2ed9e6eac0eb97f17bfebce85/drivers/md/dm-verity.c#L680-L686) which is, I guess, given by the user space for attaching/opening a new verity device.

[cryptsetup](https://gitlab.com/cryptsetup/cryptsetup/-/blob/9ae7e122ac99fc904d43dc5f2de066e1e59b6b97/src/veritysetup.c#L113) uses to `0` in the veritysetup initial commit, but it has updated it to [1](https://gitlab.com/cryptsetup/cryptsetup/-/blob/4b0b82adc559659a723b00c5b93275aa2b960591/src/veritysetup.c#L57) before verity is released in version [1.5.0](https://gitlab.com/cryptsetup/cryptsetup/-/blob/v1_5_0/src/veritysetup.c#L38).

So `1` has always been the "current" version.

However, `cryptsetup` appears to never use format/hash-type stored in the magic/super block, and it always use the one specified by the command-line utility, which is *not* the behavior of my implementation; I used the `format=` value only if the `superblock=` is set to `false` (because we needs to specify all the parameters contains in the block).

So the message appears to be kind of "correct" as long no new format/hash-type version is not released and it does not default to newer version.

Any better suggestion?

Reply:
maybe say "modern" instead of "current"

Reply:
PS: I used the message from the kernel documentation :)